File: java/leetcode/src/Task461.java
import org.junit.jupiter.api.Test;

public class Task461 {
    public int hammingDistance(int x, int y) {
        int val = x^y;
        int res = 0;
        while(val != 0){
            val = val & (val-1);
            ++res;
        }

        return res;
    }

    @Test
    void test(){
        int x = 1;
        int y = 4;
        System.out.println(hammingDistance(x,y));
    }
}

File: go/task461/task461.go
package task461

func hammingDistance(x int, y int) int {
	xor := x ^ y
	mask := 1
	res := 0
	for xor > 0 {
		if i := xor & mask; i > 0 {
			xor = xor & ^mask
			res++
		}
		mask <<= 1
	}
	return res
}

func hammingDistance2(x int, y int) int {
	res := 0
	for xor := x ^ y; xor > 0; xor &= xor - 1 {
		res += 1
	}
	return res
}
